delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/files/routes/FilesRoutes.scala

```diff
@@ -232,6 +247,30 @@ final class FilesRoutes(
       }
     }
 
+  private def filePutRequest(id: IdSegment, proj: ProjectRef): Directive1[FilePutRequest] =
+    copyFileOperation(id, proj) | updateFileOperation | createFileOperation
+
+  private def copyFileOperation(id: IdSegment, proj: ProjectRef): Directive1[FilePutRequest] =
+    (idSegmentRef(id) & parameters(
+      "targetProject".as[ProjectRef],
+      "targetStorage".as[IdSegment].?,
+      "targetTag".as[UserTag].?
+    ))
+      .tmap[FilePutRequest] { case (idRef, targetProj, targetStorage, targetTag) =>
+        FilePutRequest.Copy(FileId(idRef, proj), targetProj, targetStorage, targetTag)
+      }
+
+  private def updateFileOperation: Directive1[FilePutRequest] =
+    parameters("rev".as[Int], "storage".as[IdSegment].?, "tag".as[UserTag].?).tmap[FilePutRequest] {
+      case (rev, storage, tag) =>
+        FilePutRequest.Update(rev, storage, tag)
+    }
+
+  private def createFileOperation: Directive1[FilePutRequest] =
+    parameters("storage".as[IdSegment].?, "tag".as[UserTag].?).tmap[FilePutRequest] { case (storage, tag) =>
+      FilePutRequest.Create(storage, tag)
+    }
+
```

Comment:
I really like the idea here, but I think this is maybe hiding a little too much? I'll try to explain better IRL